pkg/providers/instanceprofile/instanceprofile.go

```diff
@@ -21,22 +21,25 @@ import (
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/iam"
 	"github.com/aws/aws-sdk-go/service/iam/iamiface"
-	"github.com/mitchellh/hashstructure/v2"
 	"github.com/patrickmn/go-cache"
 	"github.com/samber/lo"
 	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
 
-	corev1beta1 "sigs.k8s.io/karpenter/pkg/apis/v1beta1"
-
-	"github.com/aws/karpenter-provider-aws/pkg/apis/v1beta1"
 	awserrors "github.com/aws/karpenter-provider-aws/pkg/errors"
 	"github.com/aws/karpenter-provider-aws/pkg/operator/options"
 )
 
+type NodeClass interface {
```

Comment:
Why keep this interface here?

Reply:
? You have to define the interface. Not sure I get the question.

Reply:
I did; however, update the naming to be `instanceprofile.Manager` since I think it's a little broader than NodeClass

Reply:
Why defining the interface in the instance profile? why not in the apis package?

Reply:
Ah okay, `instanceprofile.Manager` more make sense